network/patton/smartnode/snmp/mode/system.pm

```diff
@@ -0,0 +1,261 @@
+                key_values => [ { name => 'memory_usage_percentage' }, { name => 'memory_usage_bytes' }, { name => 'memory_free_bytes' }, { name => 'memory_free_percentage' }, { name => 'memory_total_bytes' } ],
+                closure_custom_output => $self->can('custom_usage_output'),
+                perfdatas => [
+                    { template => '%.2f', min => 0, max => 100, unit => '%', label_extra_instance => 1 }
+                ]
+            }
+        }
+    ];
+
+    $self->{maps_counters}->{temperature} = [
+        { label => 'probe-temperature-celsius', nlabel => 'probe.temperature.celsius', set => {
+                key_values => [ { name => 'probe_temperature_celsius' }, { name => 'display' } ],
+                output_template => 'probe temperature: %.2f C',
+                perfdatas => [
+                    { template => '%.2f', min => 0, max => 100, unit => 'C', label_extra_instance => 1 }
+                ]
+            }
+        }
+    ];
+}
+
+sub new {
+    my ($class, %options) = @_;
+    my $self = $class->SUPER::new(package => __PACKAGE__, %options, force_new_perfdata => 1);
+    bless $self, $class;
+
+    $options{options}->add_options(arguments => {
+    });
+
+    return $self;
+}
+
+my $cpuEntry_oid = '.1.3.6.1.4.1.1768.100.70.10.2.1';
+my $mappingCpu = {
+    display => { oid => '.1.3.6.1.4.1.1768.100.70.10.2.1.1' }, # cpuDescr
+    cpu_utilization_current_percentage => { oid => '.1.3.6.1.4.1.1768.100.70.10.2.1.2' }, # cpuWorkloadCurrent
+    cpu_utilization_1m_percentage => { oid => '.1.3.6.1.4.1.1768.100.70.10.2.1.3' }, # cpuWorkload1MinuteAverage
+    cpu_utilization_5m_percentage => { oid => '.1.3.6.1.4.1.1768.100.70.10.2.1.4' } # cpuWorkload5MinuteAverage
+};
+
+my $memoryPoolEntry_oid = '.1.3.6.1.4.1.1768.100.70.20.2.1';
+my $mappingMemory = {
+    display => { oid => '.1.3.6.1.4.1.1768.100.70.20.2.1.1' }, # memDescr
+    memory_total_bytes => { oid => '.1.3.6.1.4.1.1768.100.70.20.2.1.2' }, # memTotalBytes
+    memory_usage_bytes => { oid => '.1.3.6.1.4.1.1768.100.70.20.2.1.3' }, # memAllocatedBytes
+    memory_free_bytes => { oid => '.1.3.6.1.4.1.1768.100.70.20.2.1.4' } # memFreeBytes
+};
+
+my $tempProbeEntry_oid = '.1.3.6.1.4.1.1768.100.70.30.2.1';
+my $mappingTemperature = {
+    display => { oid => '.1.3.6.1.4.1.1768.100.70.30.2.1.1' }, # tempProbeDescr
+    probe_temperature_celsius => { oid => '.1.3.6.1.4.1.1768.100.70.30.2.1.2' } # currentDegreesCelsius
+};
+
+sub manage_selection {
+    my ($self, %options) = @_;
+
+    my $snmp_results = $options{snmp}->get_multiple_table(oids => [
+        { oid => $cpuEntry_oid },
+        { oid => $memoryPoolEntry_oid },
+        { oid => $tempProbeEntry_oid }
+    ], nothing_quit => 1);
+
+    my $numCpu = 1;
+    my $cpuTotal = 0;
+    my $cpuEntry = $options{snmp}->map_instance(mapping => $mappingCpu, results => $snmp_results->{$cpuEntry_oid}, instance => $numCpu);
```

Comment:
the instance can start with another value than 1. It's not the right way to do the parsing.